lib/govspeak.rb

```diff
@@ -118,6 +132,26 @@ def preprocess(source)
       source
     end
 
+    def legislative_list_footnote_definitions(source)
```

Comment:
I've got some comments in the `remove_forbidden_characters` method which equally apply to this one.

But that does make me wonder, can we reuse any of the code, rather than duplicate the logic in both places? It feels like we could DRY things up by changing the internal API to something like:

```ruby
list_items = each_footnote(footnotes).map do | footnote, index|
  <<HTML_SNIPPET
    <li id="fn:#{index}" role="doc-endnote">
      <p>
        #{footnote}<a href="#fnref:#{index}" class="reversefootnote" role="doc-backlink" aria-label="go to where this is referenced">↩</a>
      </p>
    </li>
  HTML_SNIPPET
end


@footnote_definition_html = <<HTML_CONTAINER
  <div class="footnotes" role="doc-endnotes">
    <ol>
      #{list_items}
    </ol>
  </div>
HTML_CONTAINER
```

Reply:
Thanks @ChrisBAshton , this is definitely a much more elegant way of dealing with the HTML! I've updated it [here](https://github.com/alphagov/govspeak/pull/216/files#diff-acb6db6cd67285bd5380398c9e51ced8940ed7c4110706d038cbb243310e4055R146-R161)